### PR TITLE
Vinai/dolt merge ret type

### LINKED ISSUE #1937: Change result type for `dolt_merge()`
Currently `dolt_merge()` is string valued. To make it easier for clients to detect conflicts on merge we should return a boolean . `1` (true) for no conflicts, `0` (false) for conflict in merge

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Fixes #1937

### CODE REVIEW

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ type DoltMergeFunc struct {
 	expression.NaryExpression
 }
 
+const DoltConflictWarningCode int = 1105
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:40]: why `1105`?

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:40]: at least needs a comment

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go [hunk 2]
```diff
@@ -37,6 +37,8 @@ type DoltMergeFunc struct {
 	expression.NaryExpression
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:37]: the return type of this function is still `sql.Text`. line 304:
```
func (d DoltMergeFunc) Type() sql.Type {
	return sql.Text
}
```

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go [hunk 3]
```diff
@@ -88,15 +90,15 @@ func (d DoltMergeFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 
 		err := sess.SetWorkingSet(ctx, dbName, ws, nil)
 		if err != nil {
-			return nil, err
+			return 1, err
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:93]: let's create some `const`s for these return values

**Review by andrew-wm-arthur — APPROVED** [review]
Couple comments

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go (hunk 2 above)
Comment (VinaiRachakonda) [reply go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:37]: Good catch

**Pull request merged**